app/config.py

```
@@ -51,6 +52,21 @@ class Settings(BaseSettings):
+    # onnx 사용 시 scripts/build_reranker_onnx.py 산출물 디렉토리를 reranker_onnx_dir로 지정.
+    reranker_backend: Literal["torch", "onnx"] = "torch"
+    reranker_onnx_dir: str = ""
+    reranker_onnx_file: str = "model_quantized.onnx"
+
+    @model_validator(mode="after")
+    def _check_reranker_onnx(self) -> "Settings":
+        # fail-fast: onnx 백엔드인데 산출물 경로가 없으면 기동 단계에서 막는다(런타임 조용한 vector 폴백 방지).
+        if self.reranker_backend == "onnx" and not self.reranker_onnx_dir:
+            raise ValueError(
+                "reranker_backend='onnx'면 reranker_onnx_dir 필요 (scripts/build_reranker_onnx.py 산출물 경로)"
+            )
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)